fem/prestriction.hpp

```diff
@@ -88,6 +102,31 @@ class ParNCH1FaceRestriction : public H1FaceRestriction
    */
    void ComputeGatherIndices(const ElementDofOrdering ordering,
                              const FaceType type);
+
+public: // For nvcc
+   /** @brief Apply a change of basis from coarse element basis to fine element
+       basis for the coarse face dofs.
+
+       @param[in,out] x The dofs vector that needs coarse dofs to be express in
+                        term of the fine basis.
+   */
+   void NonconformingInterpolation(Vector& x) const;
+
+   /** @brief Apply a change of basis from fine element basis to coarse element
+       basis for the coarse face dofs.
+
+       @param[in] x The dofs vector that needs coarse dofs to be express in term
+                    of the coarse basis, the result is stored in x_interp.
+   */
+   void NonconformingTransposeInterpolation(const Vector& x) const;
+
+   /** @brief Apply a change of basis from fine element basis to coarse element
+       basis for the coarse face dofs.
+
+       @param[in] x The dofs vector that needs coarse dofs to be express in term
+                    of the coarse basis, the result is stored in x_interp.
+   */
+   void NonconformingTransposeInterpolation(Vector& x) const;
```

Comment:
I'm really not sure that having a non-destructive and destructive function with the same name that differ only in the const-ness of their inputs is a good idea.

Given a non-const `Vector`, the **default** behavior of `ParNCH1FaceRestriction:: NonconformingTransposeInterpolation` will be destructive, which seems dangerous.